enhancements/installer/gcp-ipi-shared-vpc.md

```diff
@@ -0,0 +1,294 @@
+- Enable users to alternatively provide pre-provisioned resources to enable
+  minimal permission requirements in the Host Project.
+
+### Non-Goals
+
+- Provisioning and/or configuring the VPC, Network, and Subnetworks are to be
+  done manually by the user prior to running the installer.
+- Provisioning and/or configuring the Service Accounts for the Installer and
+  Cluster are to be done manually by the user prior to running the installer.
+
+## Proposal
+
+The installer should be able to install a GCP IPI cluster using Shared VPC
+networking by automating the majority of the current GCP UPI workflow. This
+requires some resources to be created and configured in advance, such as the
+network and subnets. Other resources can be created in advance when the
+necessary host project permissions are not desired, such as the firewall rules
+and/or DNS configuration. The installer, and some operators, will be updated to
+enable this functionality. They will become capable of creating and managing
+resources that exist in a different (Host) project from the cluster (Service
+Project).  They will also become capable of consuming pre-existing firewall
+and/or DNS configurations.
+
+### Workflow Description
+
+Prior to executing the installer, the Shared VPC, Network, and Subnets will be
+manually created in the Host Project and configured for Shared VPC (XPN)
+configuration with the Service Project. A service account will also be created
+in the Service Project and granted sufficient permissions in the Host Project.
+When minimal permissions are desired in the Host Project, additional resources
+will be manually created and configured in the Host Project to include:
+firewall rules, api compute address, ingress compute address, and the api,
+api-int, and \*.apps DNS records.
+
+After all of these resources are created, they will be used in a manually
+created install-config.yaml to configure the cluster accordingly. At a minimum,
+the credentialsMode will be set to Passthrough, and the network,
+computeNetwork, controlPlaneNetwork, and networkProjectID will be configured to
+reference the Shared VPC network. When minimial permissions are desired in the
+Host Project, the additional parameters will be configured to enable the
+cluster to consume them as well.
+
+The installation process can then proceed as any other IPI workflow by running
+the create manifests, create ignition, and/or create cluster sub-commands.
+
+### API Extensions
+
+#### api.dns
+
+The current DNSZone struct does not appear to be able to store which project
+the zone is in. The ID variable can be used to store fully qualified path
+names, which will include the project. However, this may require modification
+of all consumers of this variable to be able to parse this information.
+Alternatively, a second variable can be added to store the projectID and the
+consumers can use that information when linking to the DNS Zone.
+
+#### api.ingress
```

Comment:
Same as above we need go snips showing how the api will look.
You might be needing to add some fields in the ingress cluster object and may be the ingress controller CR.

Reply:
This has been moved to another effort.